src/cargo/util/toml/mod.rs

```diff
@@ -1737,12 +1738,22 @@ impl TomlManifest {
             debug!("manifest has no build targets");
         }
 
-        if let Err(e) = unique_build_targets(&targets, package_root) {
-            warnings.push(format!(
-                "file found to be present in multiple \
-                 build targets: {}",
-                e
-            ));
+        if let Err(conflict_targets) = unique_build_targets(&targets, package_root) {
+            conflict_targets.iter().for_each(|c| {
```

Comment:
A suggestion on code style, I think it is usually better to give individual tuple elements a name instead of using `.0` and `.1`.  For example, here it could be something like `for_each(|(target_path, conflicts)| {` or something like that.

Reply:
Updated. Make sense!